contracts/SponsorFaucet.sol

```diff
@@ -131,95 +97,80 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard, WhitelistedRole {
 
     //set bounds for sponsorship
     function setBounds(
-        uint256 gasTotalLimit,
-        uint256 collateralTotalLimit,
-        uint256 gasBound,
-        uint256 collateralBound,
-        uint256 upperBound
+        bounds memory smallBounds
     ) public {
         _onlyWhitelistAdmin();
         //rule by internal contract
-        require(upperBound.mul(1000) <= gasBound, "upperBound too high");
-        gas_total_limit = gasTotalLimit;
-        collateral_total_limit = collateralTotalLimit;
-        gas_bound = gasBound;
-        collateral_bound = collateralBound;
-        upper_bound = upperBound;
+        require(smallBounds.upper_bound.mul(1000) <= smallBounds.gas_bound, "upperBound too high");
+        dapp_bounds[small] = smallBounds;
     }
 
     //set bounds for general contracts
     function setGeneralBounds(
-        uint256 gasTotalLimit,
-        uint256 collateralTotalLimit,
-        uint256 gasBound,
-        uint256 collateralBound,
-        uint256 upperBound
+        bounds memory largeBounds
     ) public {
         _onlyWhitelistAdmin();
         //rule by internal contract
-        require(upperBound.mul(1000) <= gasBound, "upperBound too high");
-        general_gas_total_limit = gasTotalLimit;
-        general_collateral_total_limit = collateralTotalLimit;
-        general_gas_bound = gasBound;
-        general_collateral_bound = collateralBound;
-        general_upper_bound = upperBound;
+        require(largeBounds.upper_bound.mul(1000) <= largeBounds.gas_bound, "upperBound too high");
+        dapp_bounds[large] = largeBounds;
     }
 
-    //set bounds for special contracts
-    function setSpecialBounds(
+    //set bounds for custom contracts
+    function setCustomBounds(
         address addr,
-        uint256 gasTotalLimit,
-        uint256 collateralTotalLimit,
-        uint256 gasBound,
-        uint256 collateralBound,
-        uint256 upperBound
+        bounds memory customBounds
     ) public {
         _onlyWhitelistAdmin();
         //rule by internal contract
-        require(upperBound.mul(1000) <= gasBound, "upperBound too high");
-        special_dapps[addr].gas_total_limit = gasTotalLimit;
-        special_dapps[addr].collateral_total_limit = collateralTotalLimit;
-        special_dapps[addr].gas_bound = gasBound;
-        special_dapps[addr].collateral_bound = collateralBound;
-        special_dapps[addr].upper_bound = upperBound;
+        require(customBounds.upper_bound.mul(1000) <= customBounds.gas_bound, "upperBound too high");
+        dapp_bounds[addr] = customBounds;
     }
 
 
     /* ===== Public utility functions ===== */
-    function addGeneralContracts(address[] memory addAddr) public {
+    function addLargeContracts(address[] memory addAddr) public {
         _onlyWhitelistAdmin();
         for(uint256 i = 0; i < addAddr.length; i++) {
-            general_contracts.push(addAddr[i]);
+            large_contracts[addAddr[i]] = true;
```

Comment:
Better to check if current value is `false` so as to save gas consumption.

Also, pay attention to gas consumption for all `write` function.

Reply:
fixed